lib/combine_pdf/api.rb

```diff
@@ -11,6 +11,12 @@ def load(file_name = '', options = {})
     PDF.new(PDFParser.new(IO.read(file_name, mode: 'rb').force_encoding(Encoding::ASCII_8BIT), options))
   end
 
+  def encrypted?(file_name, options = {})
```

Comment:
This approach requires parsing the file twice. Once when testing for encryption and once when actually using the file.

It makes more sense, from a design's point of view, to move the method to the `CombinePDF::PDF` object, where one could test for `parsing_errors?` as well as `encrypted?`.

Reply:
You mean it requires parsing it twice b/c you ask if it's encrypted before loading it?

Yeah, for sure I could buy into a "Tell, Don't Ask" approach to it. _Ideally_ I would rather it just raise an exception if it's encrypted since it's not supported in the first place.

Would _that_ be acceptable @boazsegev?

Reply:
Also if it wasn't clear, the _reason_ I did not put it in `PDF` is because instantiating that raises a `Zlib::DataError` for some that have that problem _and_ are encrypted, so you lose sight of the fact that the PDF was encrypted because the `Zlib::DataError` shadows it (if that makes sense).

If there weren't PII in the PDF file that has this scenario, I'd have put it in the `test/fixtures/files` but unfortunately I can't share it.

Reply:
Also, in _some_ circumstances it **does** raise this exception:

```ruby
irb(main):001:0> CombinePDF.load('11950573-78fe-49a3-bc3c-2340a9681593_files')
PDF is Encrypted! Attempting to decrypt - not yet fully supported.
Data raising exception:
 {:P=>-1324
 :R=>4
 :StrF=>:StdCF
 :CF=>{:StdCF=>{:Type=>:CryptFilter
 :CFM=>:AESV2
 :Length=>16
 :EncryptMetadata=>false}}
 :Filter=>:Standard
 :Length=>128
 :U=>"\xA7\x90\xEE\xF4\x11&r^8~\xEC\xDD\x9A\x93\x8Dr(\xBFN^Nu\x8AAd\x00NV\xFF\xFA\x01\b"
 :V=>4
 :StmF=>:StdCF
 :EncryptMetadata=>false
 :O=>"\x8B$\xBE\xE1\xC3\xDD\xB7\xAC\xF8\xC5b\x17\xA1i\xD0U\t\xC7\xD3\xF9:\x11n{\x1D\xE0@\xC6]\x96\x06{"
 :indirect_generation_number=>0
 :indirect_reference_id=>100}
File is encrypted - not supported.
File is encrypted - not supported.
/Users/levi.wilson/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bundler/gems/combine_pdf-5c4a763e45db/lib/combine_pdf/decrypt.rb:187:in `raise_encrypted_error'
/Users/levi.wilson/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bundler/gems/combine_pdf-5c4a763e45db/lib/combine_pdf/decrypt.rb:57:in `decrypt'
/Users/levi.wilson/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bundler/gems/combine_pdf-5c4a763e45db/lib/combine_pdf/parser.rb:102:in `parse'
/Users/levi.wilson/.rbenv/versions/2.6.5/lib/ruby/gems/2.6.0/bundler/gems/combine_pdf-5c4a763e45db/lib/combine_pdf/pdf_public.rb:98:in `initialize'
```

Maybe I have a misunderstanding but is `PDF is Encrypted!` different than "password protected PDF"? I ask because we probably have 3 or 4 variations of errors that happen for files that _do_ have password protections on them.

However, in _this_ example (the stack trace...again, I can't share the file) I _can_ open this file in Preview and it is **not** password protected at all, so I'm not sure why it's raising this error.

If there is something further I could look into regarding this I'd be happy to share my findings.